src/Raven.Server/Documents/Sharding/ShardedDocumentsStorage.cs

```diff
@@ -159,14 +160,32 @@ internal static ByteStringContext.Scope ExtractIdFromKeyAndGenerateBucketAndEtag
 
         var etag = *(long*)tvr.Read(etagIndex, out _);
 
-        return GenerateBucketAndEtagSlice(context, keyPtr, sizeOfDocId, etag, out slice);
+        return GenerateBucketAndEtagSlice(tx, keyPtr, sizeOfDocId, etag, out slice);
     }
 
-    private static ByteStringContext.Scope GenerateBucketAndEtagSlice(ByteStringContext context, byte* idPtr, int idSize, long etag, out Slice slice)
+    private static ByteStringContext.Scope GenerateBucketAndEtagSlice(Transaction tx, byte* idPtr, int idSize, long etag, out Slice slice)
     {
-        var scope = context.Allocate(sizeof(long) + sizeof(int), out var buffer);
+        var scope = tx.Allocator.Allocate(sizeof(long) + sizeof(int), out var buffer);
 
-        var bucket = ShardHelper.GetBucket(idPtr, idSize);
+        var bucket = ShardHelper.GetBucketFor(idPtr, idSize);
+
+        var database = tx.Owner as ShardedDocumentDatabase;
+        var prefixedConfiguration = database?.ReadShardingState().Prefixed;
+        if (prefixedConfiguration != null)
+        {
+            var idAsStr = Encoding.UTF8.GetString(idPtr, idSize);
```

Comment:
didn't we start with a string in the caller?

Reply:
no, we start we a TableValueReader and id position in value reader